www/js/control/SettingRow.jsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { angularize} from "../angular-react-helper";
+import { StyleSheet } from 'react-native';
+import { List, IconButton, Switch} from 'react-native-paper';
+import { useTranslation } from "react-i18next";
+import { string, func, bool} from "prop-types";
+
+const SettingRow = ({textKey, iconName, action, desc, switchValue}) => {
+    const { t } = useTranslation(); //this accesses the translations
+
+    let rightComponent;
+    // will using switchValue here only render when the value is true?
```

Comment:
if `switchValue` is not given, it's `undefined`; the condition will evaluate as false and the interpreter will move to the `else` block

Reply:
I've kept working with the switches, and they're consistently showing up now, but the values aren't always changing when expected - when I toggle Tracking from "off" to "on" the value shown on the switch is not changing, but it changes fine when I toggle from "on" to "off".

Reply:
I was able to make some progress with the switches using a dedicated state and useEffect() watch method, the tracking toggle now changes whenever it is clicked!